### PR TITLE
remove v0 version of the database

### PR DESCRIPTION
**Pull request by matkt**
## PR description

This version of the database is not used anymore so we can remove it

### CODE REVIEW

**Review by fab-10 — COMMENTED** [review]
LGTM, are you planning to release this in the next RC?
Please add a breaking change in the changelog

**Comment by matkt** [comment]
> LGTM, are you planning to release this in the next RC? Please add a breaking change in the changelog

yes I'm planning to add it

**Review by fab-10 — APPROVED** [review]
LGTM, just a rewording suggestion

**Review by garyschulte — APPROVED** [review]
🥳 

looks like a couple more unit tests need :eyes: though

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,8 @@
 
 ### Breaking Changes
 
+- Deprecated version 0 of the database as it is no longer used by any active nodes.
```
[the remark this answers is not in the document]
Comment (matkt) [reply CHANGELOG.md:8]: done

**Comment by matkt** [comment]
will start a sepolia node to validate this PR

File: ethereum/core/src/jmh/java/org/hyperledger/besu/ethereum/vm/operations/OperationBenchmarkHelper.java [hunk 2]
```diff
@@ -58,12 +61,18 @@ private OperationBenchmarkHelper(
 
   public static OperationBenchmarkHelper create() throws IOException {
     final Path storageDirectory = Files.createTempDirectory("benchmark");
-    final KeyValueStorage keyValueStorage =
-        new RocksDBKeyValueStorage(
+    final OptimisticRocksDBColumnarKeyValueStorage optimisticRocksDBColumnarKeyValueStorage =
+        new OptimisticRocksDBColumnarKeyValueStorage(
```
Comment (siladu) [inline ethereum/core/src/jmh/java/org/hyperledger/besu/ethereum/vm/operations/OperationBenchmarkHelper.java:65]: Struggling to see the relevance of this change...KeyValueStorage still exists doesn't it? Or maybe we should delete KeyValueStorage too?

**Review by siladu — COMMENTED** [review]
Love a bit of code deletion :)

Seems fine to me, but would like @jframe or @macfarla to confirm too please.

File: ethereum/core/src/jmh/java/org/hyperledger/besu/ethereum/vm/operations/OperationBenchmarkHelper.java (hunk 2 above)
Comment (macfarla) [reply ethereum/core/src/jmh/java/org/hyperledger/besu/ethereum/vm/operations/OperationBenchmarkHelper.java:65]: looks like the change is it's now a two step process - KeyValueStorage (which is the interface used by plugins) is created from this storage

Comment (matkt) [reply ethereum/core/src/jmh/java/org/hyperledger/besu/ethereum/vm/operations/OperationBenchmarkHelper.java:65]: yes we want to start a refactor of this part to be only segmented as we are only using that now. this one is the first PR and we will have more PRs after this one. We found a lot of code smell in this part when we wanted to change the code in order to have only one transaction to commit the worldstate

**Pull request merged**